src/ophyd_async/epics/adcore/_core_logic.py

```diff
@@ -89,34 +91,55 @@ async def set_exposure_time_and_acquire_period_if_supplied(
                 self.driver.acquire_period.set(full_frame_time, timeout=timeout),
             )
 
-    async def start_acquiring_driver_and_ensure_status(self) -> AsyncStatus:
+    async def start_acquiring_driver_and_ensure_status(
+        self, timeout: float = DEFAULT_TIMEOUT
+    ) -> AsyncStatus:
         """Start acquiring driver, raising ValueError if the detector is in a bad state.
 
         This sets driver.acquire to True, and waits for it to be True up to a timeout.
         Then, it checks that the DetectorState PV is in DEFAULT_GOOD_STATES,
         and otherwise raises a ValueError.
 
+
+        :param timeout:
+            Timeout used for waiting for the driver to start
+            acquiring, and for waiting for the detector to detector to
+            be in a good state after it stops acquiring
         :returns AsyncStatus:
             An AsyncStatus that can be awaited to set driver.acquire to True and perform
             subsequent raising (if applicable) due to detector state.
+
         """
+        t0 = time.monotonic()
         status = await set_and_wait_for_value(
             self.driver.acquire,
             True,
-            timeout=DEFAULT_TIMEOUT,
+            timeout=timeout,
             wait_for_set_completion=False,
         )
 
         async def complete_acquisition() -> None:
-            # NOTE: possible race condition here between the callback from
-            # set_and_wait_for_value and the detector state updating.
             await status
-            state = await self.driver.detector_state.get_value()
-            if state not in self.good_states:
-                raise ValueError(
-                    f"Final detector state {state.value} not "
-                    "in valid end states: {self.good_states}"
-                )
+            state = None
+            timeout_left = timeout - (time.monotonic() - t0)
```

Comment:
Hmm, nice idea to use the total timeout for both actions, but given they are separated by some period of time it probably doesn't add much to do this. On reflection then maybe we should just pass the two timeouts into the function directly:
```python
        :param start_timeout:
            Timeout used for waiting for the driver to start acquiring
        :param state_timeout:
            Timeout used for waiting for the detector to be in a good state
            after it stops acquiring
```

Reply:
Done: https://github.com/bluesky/ophyd-async/pull/889/commits/d2e5aa838b74188ac7164cff9a7d367f90f77004